lib/src/rules/avoid_dynamic_calls.dart

```diff
@@ -21,6 +21,16 @@ Additionally, targets typed "dynamic" disables most static analysis, meaning it
 is easier to lead to a runtime "NoSuchMethodError" or "NullError" than properly
 statically typed Dart code.
 
+There is an exception to methods and properties that exist on "Object?":
+- a.hashCode
+- a.runtimeType
+- a.noSuchMethod(someInvocation)
+- a.toString()
+
+... these members are dynamically dispatched in the web-based runtimes, but not
+in the VM-based ones. Additionally, they are so common that it would be very
+punishing to disallow `any.toString()` or `any == true`, for example.
```

Comment:
I think this description is misleading. AFAIU, dart2js does not distinguish between dynamic and non-dynamic calls at all (except for possibly inserting extra type checks). In DDC, the calls in
```dart
String foo1(dynamic x) => x.toString();
String foo2(Object? x) => x.toString();
String foo3(Object x) => x.toString();
```
are all translated identically, but actual dynamic calls, like
```dart
String bar1(dynamic x) => x.toString(color: "fuchsia");
String bar2(dynamic x) => x.nothing();
```
are translated differently from that (using `dart.dsend`, giving the name of the method as a string).

So the categorization of these calls as non-dynamic holds equally well across all of our compilers.

Reply:
That can't possibly be true, because `undefined.toString()` and`null.toString()` in JS is a runtime error.

Reply:
In either case I think we can tweak the description later. Merging!